src/lib/autocomplete/autocomplete-core.ts

```diff
@@ -239,12 +242,22 @@ export class AutocompleteCore extends ListDropdownAwareCore implements IAutocomp
   }
 
   private _applyBlur(): void {
+    const shouldSelectFirstOptionOnBlur = this._shouldSelectFirstOptionOnBlur();
+    if (shouldSelectFirstOptionOnBlur) {
+      this._selectOnBlurPending = true;
+    }
+
     if (this._isDropdownOpen) {
       this._closeDropdown();
     }
 
     // If we are in stateless mode, we don't need to do anything further
     if (this._mode === AutocompleteMode.Stateless) {
+      this._selectOnBlurPending = false;
+      return;
+    }
+
+    if (shouldSelectFirstOptionOnBlur) {
```

Comment:
Couldn't this be simplified?  If it's Stateless, then shouldSelectFirstOnBlur() will return false, so I'm not sure explicitly setting it is necessary.

```ts
if (this._isDropdownOpen) {
   this._closeDropdown();
}

this._selectOnBlurPending = this._shouldSelectFirstOnBlur();
if (this._selectOnBlurPending || this._mode === AutocompleteMode.Stateless) {
    return;
}
```